packages/core-data/src/hooks/use-entity-record.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * WordPress dependencies
+ */
+import { useQuerySelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as coreStore } from '../';
+import { IDLE, SUCCESS, ERROR, RESOLVING } from './constants';
+
+/**
+ * @typedef {IDLE|RESOLVING|SUCCESS|ERROR} EntityRecordStatus
```

Comment:
Why don't we write this file in typescript instead? Could be easier to read and potentially produce nicer type definitions.

Reply:
Somehow I missed the fact that we can now use .ts files – thanks!